### PR TITLE
Make sure berserk uses damage taken for dragon darts 

### PR DESCRIPTION
**Pull request by pacmanboss256**

### CODE REVIEW

**Comment by DaWoblefet** [comment]
Doesn't this solution make Dragon Darts not trigger Berserk if the 2nd hit brings it under 50%?

**Comment by pacmanboss256** [comment]
hmmmmm maybe yeah

File: data/abilities.ts [hunk 1]
```diff
@@ -341,7 +341,8 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 			if (!source || source === target || !target.hp || !move.totalDamage) return;
 			const lastAttackedBy = target.getLastAttackedBy();
 			if (!lastAttackedBy) return;
-			const damage = move.multihit ? move.totalDamage : lastAttackedBy.damage;
+			const darts = move.id === 'dragondarts' && source.getSmartTargets(target, move).length > 1;
```
Comment (Marty-D) [inline data/abilities.ts:344]: Using `getSmartTargets` here doesn't make sense; it's not necessarily going to give you the same result as when it was used before damage happened.

**Comment by pacmanboss256** [comment]
bump

**Comment by Marty-D** [comment]
Shouldn't Berserk activate in general when hit by one dart from above 50%?

**Comment by pacmanboss256** [comment]
yes, i just realized that a few minutes ago. Is there a way to check how many times a move hit a specific pokemon?

**Comment by Marty-D** [comment]
Dragon Darts is just fundamentally broken, then. In the test what you're getting out of `getLastAttackedBy().damage` is 11, which is the damage actually dealt to Togedemaru, not Drampa. Meanwhile, the damage dealt to Togedemaru is stored as 0.

**Pull request closed without merging**